### PR TITLE
[Bugfix] Fix limited access grader anonymous forum bug

### LINKED ISSUE #3523: Anonymous issues (for limited access graders)
Introduced in #3302. Limited access graders have access to student names who posted anonymously. To recreate simply open the edit post or show history option on a post that is anonymous.

### PR DESCRIPTION
**Pull request by andrewaikens87**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [x] Tests for the changes have been added/updated (if possible)
- Shouldn't be required
* [x] Documentation has been updated/added if relevant
- Shouldn't be required

### What is the current behavior?
Fixes #3523.

### What is the new behavior?
No longer displays user ids that asked to be anonymous in show history and edit post/thread for limited access graders. Editing a post for a user now has anonymous where the user id existed, for both limited and full access graders. Show history for full access graders has been left unchanged. 

### Other information?
Shouldn't break anything.
Test full access grader vs a limited access grader.

### CODE REVIEW

**Comment by bmcutler** [comment]
@shwehtom89
Please prioritize testing & reviewing the code for this patch
Thanks!

File: site/app/controllers/forum/ForumController.php [hunk 1]
```diff
@@ -775,15 +795,20 @@ public function getHistory(){
...
         $post_id = $_POST["post_id"];
         if($this->checkPostEditAccess($post_id) && !empty($post_id)) {
             $result = $this->core->getQueries()->getPost($post_id);
             $output = array();
-            $output['user'] = $result["author_user_id"];
             $output['post'] = $result["content"];
             $output['post_time'] = $result['timestamp'];
             $output['anon'] = $result['anonymous'];
+            $output['change_anon'] = $this->modifyAnonymous($result["author_user_id"]);
+            $output['user'] = $output['anon'] ? 'Anonymous' : $result["author_user_id"];
```
Comment (shwehtom89) [inline site/app/controllers/forum/ForumController.php]: If a post is marked as anonymous, should the edit banner always display anonymous? For limited access graders this should always be the case. But for the author, ta's and instructors should this display anonymous?

**Comment by andrewaikens87** [comment]
I had a typo for a param to one of the functions and made a couple more tweaks. I will go over it more tomorrow but should be testable now.

**Comment by shwehtom89** [comment]
Alright looks good. One small cosmetic thing I've noticed is the eye icon in the history popup. Clicking on it does not reveal the student's identity, but should they eye be there in the first place?
history

**Pull request merged**